dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/internal/parser/DokuWikiIterativeParser.java

```diff
@@ -958,18 +958,40 @@ private void processLink(Reader source, Listener listener) throws IOException
                             new ArrayList<>(functionBuffer
                                     .subList(1, functionBuffer.size() - 2))), ResourceType.MAILTO);
                     reference.setTyped(true);
-                } else {
-                    reference = new ResourceReference(getStringRepresentation(
-                            new ArrayList<>(functionBuffer.subList(0, functionBuffer.size() - 1))), ResourceType.URL);
-                    reference.setTyped(false);
+                    listener.beginLink(reference, false, Listener.EMPTY_PARAMETERS);
+                    processWordsFromReader((char) c, source, listener, TAG_DOUBLE_CLOSING_SQUARE_BRACKETS);
+                    listener.endLink(reference, false, Listener.EMPTY_PARAMETERS);
+                    break;
+                }
+                else {
+                    if (c != '{'){
+                        reference = new ResourceReference(getStringRepresentation(
+                                new ArrayList<>(functionBuffer.subList(0, functionBuffer.size() - 1))), ResourceType.URL);
```

Comment:
You should not assume it's always going to be a URL type here, as far as I can see in your tests it's really not the case.

Reply:
This is the "else" option part of the previous code, that handles the input if it's not mail. I didn't make any significant change here, just put it into the condition not to run for an imagelink.

Reply:
Sorry wrong line, I just copy/pasted this comment in the right place.